File: LinkedList/python_Implementation/singlyLinkedList.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

# Iterative :
def traverse_Iter(head : Node):
    current_Node = head

    while current_Node != None:
        print(current_Node.val)
        current_Node = current_Node.next

# Recursive :
def traverse_Recur(head : Node):
    if head == None:
        return

    print(head.val)
    traverse_Recur(head.next)

if __name__ == "__main__":
    a = Node('A')
    b = Node('B')
    c = Node('C')
    d = Node('D')
    e = Node('E')
    
    a.next = b
    b.next = c
    c.next = d
    d.next = e

    traverse_Iter(a)
    print()
    traverse_Recur(a)

File: LinkedList/python_Implementation/singlyLinkedList.js
class Node {
    constructor(val,next = null){
        this.val = val;
        this.next = next;
    }
}

// Iterative Traversal
const iter_Traversal = (head) => {
    let cur_Node = head;

    while (cur_Node != null){
        console.log(cur_Node.val);
        cur_Node = cur_Node.next;
    }
}

// Recursive Traversal
const recur_Traversal = (head) => {
    if (head == null){
        return
    }
    console.log(head.val);
    recur_Traversal(head.next);
}

const A = new Node('A');
const B = new Node('B');
const C = new Node('C');

A.next = B;
B.next = C;

iter_Traversal(A);
recur_Traversal(A);